sensors.py

```diff
@@ -100,21 +130,17 @@ def update(self) -> None:
             self.data += data
         frame = self.get_data(self.data)
         if frame:
-            self.pm[SensorType.PM1] = int.from_bytes(data[4:6], byteorder=self.BYTEORDER)
-            self.pm[SensorType.PM2_5] = int.from_bytes(data[6:8], byteorder=self.BYTEORDER)
-            self.pm[SensorType.PM10] = int.from_bytes(data[8:10], byteorder=self.BYTEORDER)
+            self._add_reading(SensorType.PM1, int.from_bytes(data[4:6], byteorder=self.BYTEORDER))
+            self._add_reading(SensorType.PM2_5, int.from_bytes(data[6:8], byteorder=self.BYTEORDER))
+            self._add_reading(SensorType.PM10, int.from_bytes(data[8:10], byteorder=self.BYTEORDER))
             self.data = bytearray()
 
     def get_reading(self, sensor_type: SensorType) -> int | float:
+        assert sensor_type in self._sensor_types, f"Wrong PMSA003C sensor type({sensor_type})"
         with self._lock:
             try:
                 self.update()
-            except Exception as exc:
+            except pigpio.error as exc:
                 raise SensorReadingError from exc
 
-            tmp = self.pm[sensor_type]
-            if tmp is None:
-                raise SensorReadingError
-
-            self.pm[sensor_type] = None
-            return tmp
+            return self._get_median(sensor_type)
```

Comment:
De-indent as we did earlier?
Please consider.

Reply:
Can't do this here, each time get_reading is called all PM values are updated.
Without lock it could happen during getting median

Reply:
lock is blocking by default, so inconsistent-return-statements doesn't apply here. Looks good.